readthedocs/projects/views/private.py

```diff
@@ -113,6 +114,28 @@ def get_context_data(self, **kwargs):
             # Alternatively, dynamically override super()-derived `project_list` context_data
             # context[self.get_context_object_name(filter.qs)] = filter.qs
 
+            projects = AdminPermission.projects(user=self.request.user, admin=True)
+            if (timezone.now() - self.request.user.date_joined).days < 30:
+                context[
+                    "promotion"
+                ] = "includes/elements/promotions/example-projects.html"
+            elif (
+                projects.exists()
+                and not projects.filter(external_builds_enabled=True).exists()
+            ):
+                context[
+                    "promotion"
+                ] = "includes/elements/promotions/pull-request-previews.html"
+            elif (
+                projects.exists()
+                and not projects.filter(addons__analytics_enabled=True).exists()
+            ):
+                context[
+                    "promotion"
+                ] = "includes/elements/promotions/traffic-analytics.html"
+            else:
+                context["promotion"] = "includes/elements/promotions/security-logs.html"
```

Comment:
This would be easier to read/manage with less redundancy, seems we could do:

```
if:
    template_name = "pull-request-previews.html"
context["promotion"] = f"projects/partials/dashboard/{template_name}"
```